### ISSUE TITLE
Second click on deeplink if we have already installed app then not able to read deeplink on the app

### ISSUE THREAD

**Issue by ManzzBaria** [adjust/android_sdk#397]
Hello support,
Please check below deeplink :
https://app.adjust.com/pn9kd8?deeplink=airgames%3A%2F%2Fhtmlapk%3FOP%3DBA%26campaignId%3D123%26M%3Dt%26serviceId%3D1185%26country%3DIN%26logs%3Dtrue%26load%3Daws-appsoffers.s3.ap-south-1.amazonaws.com/index.html%26source%3Dma%26src%3Dadj%26adjust_no_sdkclick%3D1

Issue 1: 
   When i click on deeplink and install app first time then i am able to read deeplink params but after uninstalled the app and again click on deeplink an install the app then i am not able to get deeplink params. but after clear device log from your dash board and click on deeplink again and install app again then i am able to reas deeplink params. 
   So here i want to read deeplink params all the time without clear device logs from your dash.

Issue 2: 
    When we have already installed the app and click on deeplink i am not able to read that deeplink params.

Can you please help me here, Its realy very very urjent for me to resolve this issue? If there is some configuration need on dash or inside app please guide me about it. (I am okay with any deeplink direct or indirect or any other, my concern is just need to read deeplink params all the time whenever click on deeplink)

Waiting for you positive responce.

Regards.

**Comment by uerceg** [comment]
Hi @ManzzBaria 

Issue 1:
This is actually expected behaviour. Thing you are trying to test in here is deferred deep linking scenario (where you're trying to get content of deep link after installing and launching an app) and that scenario takes place only after app has been installed from Adjust point of view. Installation for Adjust means first session ever incoming from a given device. Each subsequent session afterwards will be counted as simply new session. So you reinstalling an app will not change anything for us since we know about it and will not count that new session after reinstalling as an install, thus no deferred deep link being sent. In case you would like to see this scenario happen in our platform, please contact your dedicated account manager (in case you have none email to support@adjust.com should do it) and request this feature to be added.

Issue 2:
This does sound like unexpected behaviour. In case your app is installed when you click on this URL, does the app get opened? If yes, I assume you've probably set up deep linking well in your manifest file by linking `airgames` scheme with some activity which is being launched. In case that activity gets launched, where are you looking for deep link content? Make sure you're trying to get deep link content on proper place in that activity since depending on launch mode property you have set for that activity, you might need to look for it on different place(s). More info can be found in our [README](https://github.com/adjust/android_sdk#dl-standard).

Let us know in case you have any further troubles with issue 2.

Cheers

**Comment by ManzzBaria** [comment]
Hello, 
Here is my manifest file Lunch screen code:

 <activity
            android:name="com.flightgames.gamesair.SplashActivity"
            android:label="@string/app_name"
            android:launchMode="singleTask"
            android:screenOrientation="portrait"
            android:theme="@style/Theme.AppCompat.Light.DarkActionBar.NoActionBar">
            <intent-filter>
                <action android:name="android.intent.action.MAIN" />
                <category android:name="android.intent.category.LAUNCHER" />
            </intent-filter>
            <intent-filter>
                <action android:name="android.intent.action.VIEW" />
                <category android:name="android.intent.category.DEFAULT" />
                <category android:name="android.intent.category.BROWSABLE" />
                <data android:scheme="airgames" />
            </intent-filter>
        </activity>


Can you please guide me here?

**Comment by uerceg** [comment]
Hi @ManzzBaria 

I think you're expecting deep link to arrive at the same place (in your case `launchReceivedDeeplink` method you have overridden on `AdjustConfig` object) which is wrong assumption. `launchReceivedDeeplink` is a method which one can override with following purpose: Adjust SDK will deliver content of **deferred** deep link in here once we get it from backend and depending on return value you chose in this overridden method (in your case `return false;`), Adjust SDK will try to launch activity which is in charge for the scheme from deferred deep link or not (in your case we won't try to launch the activity because of `false` return value, but rather we'll just provide you with the deep link content in there as part of `deeplink` parameter).

This what I have just described is delivery of the deep link content in case of **deferred deep linking scenario**, meaning you have clicked on tracker URL with deep link parameter attached, was redirected to the store, installed an app and launched it.

When application is installed on your device and you click on tracker URL which contains deep link parameter inside, we're no longer talking about deferred deep linking scenario, but rather (not sure what's the name, we call it) standard deep linking where redirect to your app will happen upon the click and Android operating system will try to launch activity which is in charge for scheme from the deep link. In your case, it will try to launch activity called `com.flightgames.gamesair.SplashActivity`. Seems like this activity has following configured:

```xml
android:launchMode="singleTask"
```

For this kind of setting it means that after Android has attempted to open the activity it will deliver the content of the deep link which caused it's opening (if I recall well) will be delivered into `onNewIntent` method of that class (override it and try to get it, like stated in README):

```java
@Override
protected void onNewIntent(Intent intent) {
    super.onNewIntent(intent);

    Uri data = intent.getData();
    // data.toString() -> This is your deep_link parameter value.
}
```

One thing I am not sure (feel free to test it) is if this method is the one which will be called when your `singleTask` activity is launched for the first time ever from the deep link. In case you don't get deep link in that scenario in above mentioned method, try looking for it inside of `onCreate` method of the activity (also stated in README):

```java
@Override
protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_main);

    Intent intent = getIntent();
    Uri data = intent.getData();
    // data.toString() -> This is your deep_link parameter value.
}
```

And just to make it clear one more time, when it comes to **standard deep linking scenario**, you need to look for content of the link which opened your activity on your own in one of the two above mentioned methods you need to override, `launchReceivedDeeplink` method will not provide you the link in that case because that method is only used to deliver deep link content in **deferred deep linking scenario**.

In case you have any further questions, feel free to ping.

Cheers

**Comment by ManzzBaria** [comment]
The issue is resolved thanks

**Comment by Ashvini135** [comment]
HI ManzzBaria,
Please check below deeplink :
https://app.adjust.com/brrq2xo?deep_link=consumerpanelnonprod%3A%2F%2FCampaignCode%253DFBUK001%2526
CountryCode%253DUK&campaign=UK_Demo&adgroup=Demo_UK_Adgroup&creative=Demo_UK_Creative

we are trying to extract the Params CountryCode(ES) and CampaignCode(FBUK001) from this adjust Url when user click the Url first time and trying to install the app. but above link is not not work for us.

Could you please help me on this , Its big blocker for us.
if you can able to provide  some document and sample code with some sample links it really help us.

Waiting for you positive responce.

Regards.

**Comment by uerceg** [comment]
Hi @Ashvini135,

This type of information should be delivered to your app via deferred deep linking mechanism. In order to set things up for that and test it, you should do following steps:

1. Make sure that you have implemented [deferred deep link callback](https://help.adjust.com/en/article/deep-linking-android-sdk?src=search_page#deferred-deep-linking-callback) when configuring Adjust SDK in your app.
2. Make sure that tracker URL you're testing with is properly formatted (support team can help with this in case it's not, but from what I can see in your example, seems like it should be fine).
3. In case you are not using your test device ID as part of the tracker URL (which you are not in your example from above), make sure to have probabilistic matching enabled for the test tracking in the dashboard settings.
4. Uninstall your app from test device.
5. Forget your test device from our backend via testing console.
6. Click on a tracker URL from from some page you loaded in your test device browser.
7. Ignore the redirection to App/Play Store.
8. Install app on your test device and launch SDK (preferably in sandbox mode and with log level set to verbose so that you can see what's happening in the logs).
9. If all went well, you should see install being tracked + deferred deep link sent back to SDK + deferred deep linking callback you have defined being pinged by the SDK.
10. Inside of the deferred deep link callback, you should be getting content of the `deep_link` parameter from tracker URL where you can parse the content and extract parameters you are interested in.

In case you have any further questions down this road, feel free to ping.

**Comment by Ashvini135** [comment]
Hi uerceg,
1. We already gone through the steps which mentioned in https://help.adjust.com/en/article/deep-linking-android-sdk?src=search_page#deferred-deep-linking-callback

2. We asked the Support team also for checking the correct Deep link URL Format, they provide us the below link
(https://app.adjust.com/u9cr9e5?deep_link=consumerpanelnonprod%3A%2F%2F&CountryCode=ES&CampaignCode=FBES001&campaign=ES_Demo_Campaign&adgroup=ES_Demo_Adgroup&creative=ES_Demo_Creative)

3]  when I am trying to hit this link I can able to download/open the app , but  not able to fetch the param details,
in received only half part of data
![image](https://user-images.githubusercontent.com/131447537/233662517-67393226-1b14-4399-b0e3-c21a316b9685.png)

**Comment by uerceg** [comment]
Hm. Interesting. Seems like you're not getting the content of your `deep_link` parameter post scheme at all. Only thing which caught my eye by looking at your link is the leading ampersand in your `deep_link` parameter after `consumerpanelnonprod://`:
```
deep_link=consumerpanelnonprod%3A%2F%2F&CountryCode=ES&CampaignCode=FBES001&campaign=ES_Demo_Campaign&adgroup=ES_Demo_Adgroup&creative=ES_Demo_Creative
```

This part of the URL decodes into: 
```
deep_link=consumerpanelnonprod://&CountryCode=ES&CampaignCode=FBES001&campaign=ES_Demo_Campaign&adgroup=ES_Demo_Adgroup&creative=ES_Demo_Creative
```
Is that intentional or by accident? If you would remove that leading ampersand, does the full content of the `deep_link` parameter get delivered inside of the intent?

**Comment by Ashvini135** [comment]
Could you please help us to understand the step 3 "In case you are not using your test device ID as part of the tracker URL (which you are not in your example from above), make sure to have probabilistic matching enabled for the test tracking in the dashboard settings"
Could you please help us how to enable this Settings on Adjust Dashboard . could please confirm Is the ADID means deviceId.

**Comment by Ashvini135** [comment]
> Hm. Interesting. Seems like you're not getting the content of your `deep_link` parameter post scheme at all. Only thing which caught my eye by looking at your link is the leading ampersand in your `deep_link` parameter after `consumerpanelnonprod://`:
> 
> ```
> deep_link=consumerpanelnonprod%3A%2F%2F&CountryCode=ES&CampaignCode=FBES001&campaign=ES_Demo_Campaign&adgroup=ES_Demo_Adgroup&creative=ES_Demo_Creative
> ```
> 
> This part of the URL decodes into:
> 
> ```
> deep_link=consumerpanelnonprod://&CountryCode=ES&CampaignCode=FBES001&campaign=ES_Demo_Campaign&adgroup=ES_Demo_Adgroup&creative=ES_Demo_Creative
> ```
> 
> Is that intentional or by accident? If you would remove that leading ampersand, does the full content of the `deep_link` parameter get delivered inside of the intent?

No, tried with removing ampersand also it's not help us in any case . 
 Callback method "" is not getting called, which is required for defered link usecase

**Comment by uerceg** [comment]
> Could you please help us to understand the step 3 "In case you are not using your test device ID as part of the tracker URL (which you are not in your example from above), make sure to have probabilistic matching enabled for the test tracking in the dashboard settings" Could you please help us how to enable this Settings on Adjust Dashboard . could please confirm Is the ADID means deviceId.

I have just checked the settings for your tracker token from the tracker URL you shared above and that part was already properly configured.

Would it be possible for you to run the test one more time and send over the SDK verbose logs? From your screenshot seems like you might be using Unity app (?), but the logs I need are native Android ones. In case you don't know how to get them, you can just run `adb logcat -s Adjust` while running the test and it should output all the native Adjust SDK logs.

**Comment by Ashvini135** [comment]
04-22 21:19:37.069 20947 21646 V Adjust  : Foreground timer fired
04-22 21:19:37.072 20947 21020 D Adjust  : Wrote Activity state: ec:3 sc:1 ssc:3 sl:803.4 ts:780.8 la:11:12:13 uuid:ef1bbf79-2289-4a2b-82ad-bbe86c36383d
04-22 21:20:37.070 20947 21646 V Adjust  : Foreground timer fired
04-22 21:20:37.075 20947 21020 D Adjust  : Wrote Activity state: ec:3 sc:1 ssc:3 sl:863.4 ts:840.8 la:11:12:13 uuid:ef1bbf79-2289-4a2b-82ad-bbe86c36383d
04-22 21:21:37.071 20947 21646 V Adjust  : Foreground timer fired
04-22 21:21:37.072 20947 21020 D Adjust  : Wrote Activity state: ec:3 sc:1 ssc:3 sl:923.4 ts:900.8 la:11:12:13 uuid:ef1bbf79-2289-4a2b-82ad-bbe86c36383d
04-22 21:22:37.071 20947 21646 V Adjust  : Foreground timer fired
04-22 21:22:37.076 20947 21020 D Adjust  : Wrote Activity state: ec:3 sc:1 ssc:3 sl:983.4 ts:960.8 la:11:12:13 uuid:ef1bbf79-2289-4a2b-82ad-bbe86c36383d
04-22 21:23:42.092 25451 25451 W Adjust  : Adjust not initialized, but resetting session callback parameters saved for launch
04-22 21:23:42.093 25451 25451 W Adjust  : Adjust not initialized, but resetting session partner parameters saved for launch
04-22 21:23:42.095 25451 25451 W Adjust  : SANDBOX: Adjust is running in Sandbox mode. Use this setting for testing. Don't forget to set the environment to `production` before publishing!
04-22 21:23:42.097 25451 25451 W Adjust  : SANDBOX: Adjust is running in Sandbox mode. Use this setting for testing. Don't forget to set the environment to `production` before publishing!
04-22 21:23:42.101 25451 25451 D Adjust  : GoogleAdId being read in the foreground
04-22 21:23:42.107 25451 25521 D Adjust  : Read Attribution: tt:51dilf6 tn:Organic net:Organic cam: adg: cre: cl: adid:e4b48e1722dddcc9c625d8d78cfef2df ct: ca:NaN cc:
04-22 21:23:42.111 25451 25521 D Adjust  : Read Activity state: ec:3 sc:1 ssc:3 sl:983.4 ts:960.8 la:11:12:13 uuid:ef1bbf79-2289-4a2b-82ad-bbe86c36383d
04-22 21:23:42.112 25451 25521 D Adjust  : Read Session Callback parameters: null
04-22 21:23:42.114 25451 25521 D Adjust  : Read Session Partner parameters: null
04-22 21:23:42.114 25451 25521 D Adjust  : adjust_config.properties file not found in this app
04-22 21:23:47.614 25451 25524 D Adjust  : GoogleAdId read 52f3897a-8f5c-4a7c-9691-9461ca24a8b0
04-22 21:23:47.626 25451 25521 I Adjust  : Google Play Services Advertising ID read correctly at start time
04-22 21:23:47.627 25451 25521 V Adjust  : Foreground timer configured to fire after 60.0 seconds of starting and cycles every 60.0 seconds
04-22 21:23:47.627 25451 25521 I Adjust  : Send in background configured
04-22 21:23:47.636 25451 25647 D Adjust  : Read Package queue: []
04-22 21:23:47.636 25451 25647 D Adjust  : Package handler read 0 packages
04-22 21:23:47.640 25451 25521 W Adjust  : Session Callback parameters are not set
04-22 21:23:47.641 25451 25521 D Adjust  : Wrote Session Callback parameters: null
04-22 21:23:47.641 25451 25521 W Adjust  : Session Partner parameters are not set
04-22 21:23:47.642 25451 25521 D Adjust  : Wrote Session Partner parameters: null
04-22 21:23:47.643 25451 25521 V Adjust  : Background timer canceled
04-22 21:23:47.643 25451 25521 V Adjust  : Foreground timer starting
04-22 21:23:47.644 25451 25521 V Adjust  : Subsession start
04-22 21:23:47.648 25451 25521 V Adjust  : Started subsession 4 of session 1
04-22 21:23:47.651 25451 25521 D Adjust  : Wrote Activity state: ec:3 sc:1 ssc:4 sl:1054.0 ts:960.8 la:11:12:13 uuid:ef1bbf79-2289-4a2b-82ad-bbe86c36383d
04-22 21:23:47.663 25451 25521 V Adjust  : Background timer canceled
04-22 21:23:47.663 25451 25521 V Adjust  : Foreground timer is already started
04-22 21:23:47.663 25451 25521 V Adjust  : Subsession start
04-22 21:23:47.665 25451 25521 V Adjust  : Time span since last activity too short for a new subsession
04-22 21:23:47.665 25451 25521 V Adjust  : Background timer canceled
04-22 21:23:47.665 25451 25521 V Adjust  : Foreground timer is already started
04-22 21:23:47.665 25451 25521 V Adjust  : Subsession start
04-22 21:23:47.666 25451 25521 V Adjust  : Time span since last activity too short for a new subsession
04-22 21:23:47.759 25451 25669 D Adjust  : InstallReferrer invoke method name: onInstallReferrerSetupFinished
04-22 21:23:47.759 25451 25669 D Adjust  : InstallReferrer invoke arg: 0
04-22 21:23:47.797 25451 25669 D Adjust  : installReferrer: utm_source=google-play&utm_medium=organic, clickTime: 0, installBeginTime: 1682003825
04-22 21:23:47.798 25451 25669 D Adjust  : installVersion: null, clickTimeServer: -1, installBeginServer: -1, googlePlayInstant: false
04-22 21:23:47.798 25451 25669 D Adjust  : Install Referrer read successfully. Closing connection
04-22 21:23:47.824 25451 25669 D Adjust  : Install Referrer API connection closed
04-22 21:23:48.269 25451 25647 D Adjust  : Added package 1 (event'7d9bdf')
04-22 21:23:48.270 25451 25521 D Adjust  : Wrote Activity state: ec:4 sc:1 ssc:4 sl:1054.6 ts:961.4 la:11:12:13 uuid:ef1bbf79-2289-4a2b-82ad-bbe86c36383d
04-22 21:23:48.271 25451 25647 V Adjust  : Path:      /event
04-22 21:23:48.271 25451 25647 V Adjust  : ClientSdk: xamarin4.29.0@android4.28.7
04-22 21:23:48.271 25451 25647 V Adjust  : Parameters:
04-22 21:23:48.271 25451 25647 V Adjust  :      android_uuid     ef1bbf79-2289-4a2b-82ad-bbe86c36383d
04-22 21:23:48.271 25451 25647 V Adjust  :      api_level        28
04-22 21:23:48.271 25451 25647 V Adjust  :      app_token        8b42ihz51shs
04-22 21:23:48.271 25451 25647 V Adjust  :      app_version      2.89.08
04-22 21:23:48.271 25451 25647 V Adjust  :      attribution_deeplink 1
04-22 21:23:48.271 25451 25647 V Adjust  :      connectivity_type 1
04-22 21:23:48.271 25451 25647 V Adjust  :      country          AU
04-22 21:23:48.271 25451 25647 V Adjust  :      cpu_type         armeabi-v7a
04-22 21:23:48.271 25451 25647 V Adjust  :      created_at       2023-04-22T21:23:48.235Z+0530
04-22 21:23:48.271 25451 25647 V Adjust  :      device_manufacturer samsung
04-22 21:23:48.271 25451 25647 V Adjust  :      device_name      SM-J701F
04-22 21:23:48.271 25451 25647 V Adjust  :      device_type      phone
04-22 21:23:48.271 25451 25647 V Adjust  :      display_height   1280
04-22 21:23:48.271 25451 25647 V Adjust  :      display_width    720
04-22 21:23:48.271 25451 25647 V Adjust  :      environment      sandbox
04-22 21:23:48.271 25451 25647 V Adjust  :      event_buffering_enabled 0
04-22 21:23:48.271 25451 25647 V Adjust  :      event_count      4
04-22 21:23:48.271 25451 25647 V Adjust  :      event_token      7d9bdf
04-22 21:23:48.271 25451 25647 V Adjust  :      gps_adid         52f3897a-8f5c-4a7c-9691-9461ca24a8b0
04-22 21:23:48.271 25451 25647 V Adjust  :      gps_adid_attempt 1
04-22 21:23:48.271 25451 25647 V Adjust  :      gps_adid_src     service
04-22 21:23:48.271 25451 25647 V Adjust  :      hardware_name    PPR1.180610.011.J701FDDUACUG2
04-22 21:23:48.271 25451 25647 V Adjust  :      language         en
04-22 21:23:48.271 25451 25647 V Adjust  :      mcc              404
04-22 21:23:48.271 25451 25647 V Adjust  :      mnc              90
04-22 21:23:48.271 25451 25647 V Adjust  :      needs_response_details 1
04-22 21:23:48.271 25451 25647 V Adjust  :      network_type     13
04-22 21:23:48.271 25451 25647 V Adjust  :      os_build         PPR1.180610.011
04-22 21:23:48.271 25451 25647 V Adjust  :      os_name          android
04-22 21:23:48.271 25451 25647 V Adjust  :      os_version       9
04-22 21:23:48.271 25451 25647 V Adjust  :      package_name     com.nielsen.consumerpanel
04-22 21:23:48.271 25451 25647 V Adjust  :      screen_density   high
04-22 21:23:48.271 25451 25647 V Adjust  :      screen_format    long
04-22 21:23:48.271 25451 25647 V Adjust  :      screen_size      normal
04-22 21:23:48.271 25451 25647 V Adjust  :      session_count    1
04-22 21:23:48.271 25451 25647 V Adjust  :      session_length   1055
04-22 21:23:48.271 25451 25647 V Adjust  :      subsession_count 4
04-22 21:23:48.271 25451 25647 V Adjust  :      time_spent       961
04-22 21:23:48.271 25451 25647 V Adjust  :      tracking_enabled 1
04-22 21:23:48.271 25451 25647 V Adjust  :      ui_mode          1
04-22 21:23:48.271 25451 25521 V Adjust  : Foreground timer suspended with 59.4 seconds left
04-22 21:23:48.271 25451 25521 V Adjust  : Background timer canceled
04-22 21:23:48.272 25451 25521 V Adjust  : Background timer starting. Launching in 60.0 seconds
04-22 21:23:48.273 25451 25521 V Adjust  : Subsession end
04-22 21:23:48.274 25451 25521 D Adjust  : Wrote Activity state: ec:4 sc:1 ssc:4 sl:1054.6 ts:961.4 la:11:12:13 uuid:ef1bbf79-2289-4a2b-82ad-bbe86c36383d
04-22 21:23:48.275 25451 25521 V Adjust  : Foreground timer is already suspended
04-22 21:23:48.275 25451 25521 V Adjust  : Subsession end
04-22 21:23:48.276 25451 25521 D Adjust  : Wrote Activity state: ec:4 sc:1 ssc:4 sl:1054.6 ts:961.5 la:11:12:13 uuid:ef1bbf79-2289-4a2b-82ad-bbe86c36383d
04-22 21:23:48.276 25451 25647 D Adjust  : Wrote Package queue: [event'7d9bdf']
04-22 21:23:48.276 25451 25647 D Adjust  : Package handler wrote 1 packages
04-22 21:23:48.276 25451 25521 V Adjust  : Foreground timer is already suspended
04-22 21:23:48.276 25451 25521 V Adjust  : Subsession end
04-22 21:23:48.284 25451 25521 D Adjust  : Wrote Activity state: ec:4 sc:1 ssc:4 sl:1054.6 ts:961.5 la:11:12:13 uuid:ef1bbf79-2289-4a2b-82ad-bbe86c36383d
04-22 21:23:48.287 25451 25701 D Adjust  : Making request to url : https://app.adjust.com/event
04-22 21:23:48.762 25451 25521 V Adjust  : Background timer canceled
04-22 21:23:48.762 25451 25521 V Adjust  : Foreground timer starting
04-22 21:23:48.763 25451 25521 V Adjust  : Subsession start
04-22 21:23:48.763 25451 25521 V Adjust  : Time span since last activity too short for a new subsession
04-22 21:23:48.763 25451 25521 V Adjust  : Background timer canceled
04-22 21:23:48.763 25451 25521 V Adjust  : Foreground timer is already started
04-22 21:23:48.763 25451 25521 V Adjust  : Subsession start
04-22 21:23:48.764 25451 25521 V Adjust  : Time span since last activity too short for a new subsession
04-22 21:23:48.765 25451 25647 V Adjust  : Package handler is already sending
04-22 21:23:48.765 25451 25647 V Adjust  : Package handler is already sending
04-22 21:23:49.831 25451 25701 D Adjust  : Response string: {"timestamp":"2023-04-22T15:53:51.540Z+0000","message":"Event request failed (Ignoring event, earlier unique event tracked)","error":"Event request failed (Ignoring event, earlier unique event tracked)"}
04-22 21:23:49.833 25451 25701 E Adjust  : Response message: Event request failed (Ignoring event, earlier unique event tracked)
04-22 21:23:49.833 25451 25701 D Adjust  : Will not retry with current url strategy
04-22 21:23:49.833 25451 25701 D Adjust  : Got response in PackageHandler
04-22 21:23:49.834 25451 25521 D Adjust  : Launching failed event tracking listener
04-22 21:23:49.834 25451 25647 D Adjust  : Wrote Package queue: []
04-22 21:23:49.835 25451 25647 D Adjust  : Package handler wrote 0 packages
04-22 21:23:49.835 25451 25647 V Adjust  : Package handler can send
04-22 21:23:51.087 25451 25521 V Adjust  : Foreground timer suspended with 57.0 seconds left
04-22 21:23:51.087 25451 25521 V Adjust  : Background timer canceled
04-22 21:23:51.088 25451 25521 V Adjust  : Background timer starting. Launching in 60.0 seconds
04-22 21:23:51.088 25451 25521 V Adjust  : Subsession end
04-22 21:23:51.089 25451 25521 D Adjust  : Wrote Activity state: ec:4 sc:1 ssc:4 sl:1057.5 ts:964.3 la:11:12:13 uuid:ef1bbf79-2289-4a2b-82ad-bbe86c36383d
04-22 21:23:51.089 25451 25521 V Adjust  : Foreground timer is already suspended
04-22 21:23:51.089 25451 25521 V Adjust  : Subsession end
04-22 21:23:51.090 25451 25521 D Adjust  : Wrote Activity state: ec:4 sc:1 ssc:4 sl:1057.5 ts:964.3 la:11:12:13 uuid:ef1bbf79-2289-4a2b-82ad-bbe86c36383d
04-22 21:24:02.999 25451 25521 V Adjust  : Url to parse (consumerpanelnonprod://&campaign=Facebook?adjust_reftag=c9uzkB51Xabkt)
04-22 21:24:03.175 25451 25521 V Adjust  : Background timer canceled
04-22 21:24:03.175 25451 25648 D Adjust  : Added sdk_click 1
04-22 21:24:03.175 25451 25521 V Adjust  : Foreground timer starting
04-22 21:24:03.175 25451 25521 V Adjust  : Subsession start
04-22 21:24:03.176 25451 25521 V Adjust  : Started subsession 5 of session 1
04-22 21:24:03.177 25451 25521 D Adjust  : Wrote Activity state: ec:4 sc:1 ssc:5 sl:1069.5 ts:964.3 la:11:12:13 uuid:ef1bbf79-2289-4a2b-82ad-bbe86c36383d
04-22 21:24:03.177 25451 25648 V Adjust  : Path:      /sdk_click
04-22 21:24:03.177 25451 25648 V Adjust  : ClientSdk: xamarin4.29.0@android4.28.7
04-22 21:24:03.177 25451 25648 V Adjust  : Parameters:
04-22 21:24:03.177 25451 25648 V Adjust  :      android_uuid     ef1bbf79-2289-4a2b-82ad-bbe86c36383d
04-22 21:24:03.177 25451 25648 V Adjust  :      api_level        28
04-22 21:24:03.177 25451 25648 V Adjust  :      app_token        8b42ihz51shs
04-22 21:24:03.177 25451 25648 V Adjust  :      app_version      2.89.08
04-22 21:24:03.177 25451 25648 V Adjust  :      attribution_deeplink 1
04-22 21:24:03.177 25451 25648 V Adjust  :      click_time       2023-04-22T21:24:02.998Z+0530
04-22 21:24:03.177 25451 25648 V Adjust  :      connectivity_type 1
04-22 21:24:03.177 25451 25648 V Adjust  :      country          AU
04-22 21:24:03.177 25451 25648 V Adjust  :      cpu_type         armeabi-v7a
04-22 21:24:03.177 25451 25648 V Adjust  :      created_at       2023-04-22T21:24:03.000Z+0530
04-22 21:24:03.177 25451 25648 V Adjust  :      deeplink         consumerpanelnonprod://&campaign=Facebook?adjust_reftag=c9uzkB51Xabkt
04-22 21:24:03.177 25451 25648 V Adjust  :      device_manufacturer samsung
04-22 21:24:03.177 25451 25648 V Adjust  :      device_name      SM-J701F
04-22 21:24:03.177 25451 25648 V Adjust  :      device_type      phone
04-22 21:24:03.177 25451 25648 V Adjust  :      display_height   1280
04-22 21:24:03.177 25451 25648 V Adjust  :      display_width    720
04-22 21:24:03.177 25451 25648 V Adjust  :      environment      sandbox
04-22 21:24:03.177 25451 25648 V Adjust  :      event_buffering_enabled 0
04-22 21:24:03.177 25451 25648 V Adjust  :      gps_adid         52f3897a-8f5c-4a7c-9691-9461ca24a8b0
04-22 21:24:03.177 25451 25648 V Adjust  :      gps_adid_attempt 1
04-22 21:24:03.177 25451 25648 V Adjust  :      gps_adid_src     service
04-22 21:24:03.177 25451 25648 V Adjust  :      hardware_name    PPR1.180610.011.J701FDDUACUG2
04-22 21:24:03.177 25451 25648 V Adjust  :      installed_at     2023-04-22T21:05:39.010Z+0530
04-22 21:24:03.177 25451 25648 V Adjust  :      language         en
04-22 21:24:03.177 25451 25648 V Adjust  :      last_interval    12
04-22 21:24:03.177 25451 25648 V Adjust  :      mcc              404
04-22 21:24:03.177 25451 25648 V Adjust  :      mnc              90
04-22 21:24:03.177 25451 25648 V Adjust  :      needs_response_details 1
04-22 21:24:03.177 25451 25648 V Adjust  :      network_type     13
04-22 21:24:03.177 25451 25648 V Adjust  :      os_build         PPR1.180610.011
04-22 21:24:03.177 25451 25648 V Adjust  :      os_name          android
04-22 21:24:03.177 25451 25648 V Adjust  :      os_version       9
04-22 21:24:03.177 25451 25648 V Adjust  :      package_name     com.nielsen.consumerpanel
04-22 21:24:03.177 25451 25648 V Adjust  :      reftag           c9uzkB51Xabkt
04-22 21:24:03.177 25451 25648 V Adjust  :      screen_density   high
04-22 21:24:03.177 25451 25648 V Adjust  :      screen_format    long
04-22 21:24:03.177 25451 25648 V Adjust  :      screen_size      normal
04-22 21:24:03.177 25451 25648 V Adjust  :      session_count    1
04-22 21:24:03.177 25451 25648 V Adjust  :      session_length   1057
04-22 21:24:03.177 25451 25648 V Adjust  :      source           deeplink
04-22 21:24:03.177 25451 25648 V Adjust  :      subsession_count 4
04-22 21:24:03.177 25451 25648 V Adjust  :      time_spent       964
04-22 21:24:03.177 25451 25648 V Adjust  :      tracking_enabled 1
04-22 21:24:03.177 25451 25648 V Adjust  :      ui_mode          1
04-22 21:24:03.177 25451 25648 V Adjust  :      updated_at       2023-04-22T21:23:27.755Z+0530
04-22 21:24:03.178 25451 25521 D Adjust  : Should not try to read Install referrer
04-22 21:24:03.178 25451 25648 D Adjust  : Making request to url : https://app.adjust.com/sdk_click
04-22 21:24:03.181 25451 25521 V Adjust  : Background timer canceled
04-22 21:24:03.181 25451 25521 V Adjust  : Foreground timer is already started
04-22 21:24:03.181 25451 25521 V Adjust  : Subsession start
04-22 21:24:03.182 25451 25521 V Adjust  : Time span since last activity too short for a new subsession
04-22 21:24:03.182 25451 25521 V Adjust  : Background timer canceled
04-22 21:24:03.183 25451 25521 V Adjust  : Foreground timer is already started
04-22 21:24:03.183 25451 25521 V Adjust  : Subsession start
04-22 21:24:03.184 25451 25521 V Adjust  : Time span since last activity too short for a new subsession
04-22 21:24:03.415 25451 25647 D Adjust  : Added package 1 (event'7d9bdf')
04-22 21:24:03.417 25451 25647 V Adjust  : Path:      /event
04-22 21:24:03.417 25451 25647 V Adjust  : ClientSdk: xamarin4.29.0@android4.28.7
04-22 21:24:03.417 25451 25647 V Adjust  : Parameters:
04-22 21:24:03.417 25451 25647 V Adjust  :      android_uuid     ef1bbf79-2289-4a2b-82ad-bbe86c36383d
04-22 21:24:03.417 25451 25647 V Adjust  :      api_level        28
04-22 21:24:03.417 25451 25647 V Adjust  :      app_token        8b42ihz51shs
04-22 21:24:03.417 25451 25647 V Adjust  :      app_version      2.89.08
04-22 21:24:03.417 25451 25647 V Adjust  :      attribution_deeplink 1
04-22 21:24:03.417 25451 25647 V Adjust  :      connectivity_type 1
04-22 21:24:03.417 25451 25647 V Adjust  :      country          AU
04-22 21:24:03.417 25451 25647 V Adjust  :      cpu_type         armeabi-v7a
04-22 21:24:03.417 25451 25647 V Adjust  :      created_at       2023-04-22T21:24:03.267Z+0530
04-22 21:24:03.417 25451 25647 V Adjust  :      device_manufacturer samsung
04-22 21:24:03.417 25451 25647 V Adjust  :      device_name      SM-J701F
04-22 21:24:03.417 25451 25647 V Adjust  :      device_type      phone
04-22 21:24:03.417 25451 25647 V Adjust  :      display_height   1280
04-22 21:24:03.417 25451 25647 V Adjust  :      display_width    720
04-22 21:24:03.417 25451 25647 V Adjust  :      environment      sandbox
04-22 21:24:03.417 25451 25647 V Adjust  :      event_buffering_enabled 0
04-22 21:24:03.417 25451 25647 V Adjust  :      event_count      5
04-22 21:24:03.417 25451 25647 V Adjust  :      event_token      7d9bdf
04-22 21:24:03.417 25451 25647 V Adjust  :      gps_adid         52f3897a-8f5c-4a7c-9691-9461ca24a8b0
04-22 21:24:03.417 25451 25647 V Adjust  :      gps_adid_attempt 1
04-22 21:24:03.417 25451 25647 V Adjust  :      gps_adid_src     service
04-22 21:24:03.417 25451 25647 V Adjust  :      hardware_name    PPR1.180610.011.J701FDDUACUG2
04-22 21:24:03.417 25451 25647 V Adjust  :      language         en
04-22 21:24:03.417 25451 25647 V Adjust  :      mcc              404
04-22 21:24:03.417 25451 25647 V Adjust  :      mnc              90
04-22 21:24:03.417 25451 25647 V Adjust  :      needs_response_details 1
04-22 21:24:03.417 25451 25647 V Adjust  :      network_type     13
04-22 21:24:03.417 25451 25647 V Adjust  :      os_build         PPR1.180610.011
04-22 21:24:03.417 25451 25647 V Adjust  :      os_name          android
04-22 21:24:03.417 25451 25647 V Adjust  :      os_version       9
04-22 21:24:03.417 25451 25647 V Adjust  :      package_name     com.nielsen.consumerpanel
04-22 21:24:03.417 25451 25647 V Adjust  :      screen_density   high
04-22 21:24:03.417 25451 25647 V Adjust  :      screen_format    long
04-22 21:24:03.417 25451 25647 V Adjust  :      screen_size      normal
04-22 21:24:03.417 25451 25647 V Adjust  :      session_count    1
04-22 21:24:03.417 25451 25647 V Adjust  :      session_length   1070
04-22 21:24:03.417 25451 25647 V Adjust  :      subsession_count 5
04-22 21:24:03.417 25451 25647 V Adjust  :      time_spent       964
04-22 21:24:03.417 25451 25647 V Adjust  :      tracking_enabled 1
04-22 21:24:03.417 25451 25647 V Adjust  :      ui_mode          1
04-22 21:24:03.418 25451 25647 D Adjust  : Wrote Package queue: [event'7d9bdf']
04-22 21:24:03.419 25451 25647 D Adjust  : Package handler wrote 1 packages
04-22 21:24:03.420 25451 25521 V Adjust  : Background timer canceled
04-22 21:24:03.420 25451 25521 V Adjust  : Background timer starting. Launching in 60.0 seconds
04-22 21:24:03.421 25451 25521 D Adjust  : Wrote Activity state: ec:5 sc:1 ssc:5 sl:1069.6 ts:964.4 la:11:12:13 uuid:ef1bbf79-2289-4a2b-82ad-bbe86c36383d
04-22 21:24:03.422 25451 25521 V Adjust  : Foreground timer suspended with 56.8 seconds left
04-22 21:24:03.422 25451 25521 V Adjust  : Subsession end
04-22 21:24:03.422 25451 25701 D Adjust  : Making request to url : https://app.adjust.com/event
04-22 21:24:03.423 25451 25521 D Adjust  : Wrote Activity state: ec:5 sc:1 ssc:5 sl:1069.8 ts:964.5 la:11:12:13 uuid:ef1bbf79-2289-4a2b-82ad-bbe86c36383d
04-22 21:24:03.423 25451 25521 V Adjust  : Foreground timer is already suspended
04-22 21:24:03.423 25451 25521 V Adjust  : Subsession end
04-22 21:24:03.425 25451 25521 D Adjust  : Wrote Activity state: ec:5 sc:1 ssc:5 sl:1069.8 ts:964.5 la:11:12:13 uuid:ef1bbf79-2289-4a2b-82ad-bbe86c36383d
04-22 21:24:03.425 25451 25521 V Adjust  : Foreground timer is already suspended
04-22 21:24:03.425 25451 25521 V Adjust  : Subsession end
04-22 21:24:03.426 25451 25521 D Adjust  : Wrote Activity state: ec:5 sc:1 ssc:5 sl:1069.8 ts:964.5 la:11:12:13 uuid:ef1bbf79-2289-4a2b-82ad-bbe86c36383d
04-22 21:24:03.544 25451 25521 V Adjust  : Background timer canceled
04-22 21:24:03.544 25451 25521 V Adjust  : Foreground timer starting
04-22 21:24:03.544 25451 25521 V Adjust  : Subsession start
04-22 21:24:03.545 25451 25521 V Adjust  : Time span since last activity too short for a new subsession
04-22 21:24:03.548 25451 25647 V Adjust  : Package handler is already sending
04-22 21:24:03.548 25451 25521 V Adjust  : Background timer canceled
04-22 21:24:03.549 25451 25521 V Adjust  : Foreground timer is already started
04-22 21:24:03.549 25451 25521 V Adjust  : Subsession start
04-22 21:24:03.550 25451 25521 V Adjust  : Time span since last activity too short for a new subsession
04-22 21:24:03.550 25451 25647 V Adjust  : Package handler is already sending
04-22 21:24:04.373 25451 25648 D Adjust  : Response string: {"app_token":"8b42ihz51shs","adid":"e4b48e1722dddcc9c625d8d78cfef2df","timestamp":"2023-04-22T15:54:06.347Z+0000","message":"Click tracked"}
04-22 21:24:04.374 25451 25648 I Adjust  : Response message: Click tracked
04-22 21:24:04.374 25451 25648 D Adjust  : Will not retry with current url strategy
04-22 21:24:04.376 25451 25521 D Adjust  : Wrote Activity state: ec:5 sc:1 ssc:5 sl:1069.8 ts:964.5 la:11:12:13 uuid:ef1bbf79-2289-4a2b-82ad-bbe86c36383d
04-22 21:24:04.492 25451 25701 D Adjust  : Response string: {"timestamp":"2023-04-22T15:54:06.475Z+0000","message":"Event request failed (Ignoring event, earlier unique event tracked)","error":"Event request failed (Ignoring event, earlier unique event tracked)"}
04-22 21:24:04.493 25451 25701 E Adjust  : Response message: Event request failed (Ignoring event, earlier unique event tracked)
04-22 21:24:04.493 25451 25701 D Adjust  : Will not retry with current url strategy
04-22 21:24:04.493 25451 25701 D Adjust  : Got response in PackageHandler
04-22 21:24:04.494 25451 25521 D Adjust  : Launching failed event tracking listener
04-22 21:24:04.494 25451 25647 D Adjust  : Wrote Package queue: []
04-22 21:24:04.495 25451 25647 D Adjust  : Package handler wrote 0 packages
04-22 21:24:04.495 25451 25647 V Adjust  : Package handler can send
04-22 21:24:08.145 25451 25521 V Adjust  : Foreground timer suspended with 52.2 seconds left
04-22 21:24:08.145 25451 25521 V Adjust  : Background timer canceled
04-22 21:24:08.145 25451 25521 V Adjust  : Background timer starting. Launching in 60.0 seconds
04-22 21:24:08.145 25451 25521 V Adjust  : Subsession end
04-22 21:24:08.146 25451 25521 D Adjust  : Wrote Activity state: ec:5 sc:1 ssc:5 sl:1074.5 ts:969.2 la:11:12:13 uuid:ef1bbf79-2289-4a2b-82ad-bbe86c36383d
04-22 21:24:08.147 25451 25521 V Adjust  : Foreground timer is already suspended
04-22 21:24:08.147 25451 25521 V Adjust  : Subsession end
04-22 21:24:08.148 25451 25521 D Adjust  : Wrote Activity state: ec:5 sc:1 ssc:5 sl:1074.5 ts:969.2 la:11:12:13 uuid:ef1bbf79-2289-4a2b-82ad-bbe86c36383d
04-22 21:24:11.342 25451 25521 V Adjust  : Background timer canceled
04-22 21:24:11.342 25451 25521 V Adjust  : Foreground timer starting
04-22 21:24:11.342 25451 25521 V Adjust  : Subsession start
04-22 21:24:11.342 25451 25521 V Adjust  : Started subsession 6 of session 1
04-22 21:24:11.343 25451 25521 D Adjust  : Wrote Activity state: ec:5 sc:1 ssc:6 sl:1077.7 ts:969.2 la:11:12:13 uuid:ef1bbf79-2289-4a2b-82ad-bbe86c36383d
04-22 21:24:11.344 25451 25521 D Adjust  : Should not try to read Install referrer
04-22 21:24:11.345 25451 25521 V Adjust  : Background timer canceled
04-22 21:24:11.345 25451 25521 V Adjust  : Foreground timer is already started
04-22 21:24:11.345 25451 25521 V Adjust  : Subsession start
04-22 21:24:11.346 25451 25521 V Adjust  : Time span since last activity too short for a new subsession
04-22 21:24:12.520 25451 25521 V Adjust  : Foreground timer suspended with 51.0 seconds left
04-22 21:24:12.520 25451 25521 V Adjust  : Background timer canceled
04-22 21:24:12.520 25451 25521 V Adjust  : Background timer starting. Launching in 60.0 seconds
04-22 21:24:12.520 25451 25521 V Adjust  : Subsession end
04-22 21:24:12.521 25451 25521 D Adjust  : Wrote Activity state: ec:5 sc:1 ssc:6 sl:1078.9 ts:970.4 la:11:12:13 uuid:ef1bbf79-2289-4a2b-82ad-bbe86c36383d
04-22 21:24:12.522 25451 25521 V Adjust  : Foreground timer is already suspended
04-22 21:24:12.522 25451 25521 V Adjust  : Subsession end
04-22 21:24:12.523 25451 25521 D Adjust  : Wrote Activity state: ec:5 sc:1 ssc:6 sl:1078.9 ts:970.4 la:11:12:13 uuid:ef1bbf79-2289-4a2b-82ad-bbe86c36383d
04-22 21:24:32.116 26292 26292 W Adjust  : Adjust not initialized, but resetting session callback parameters saved for launch
04-22 21:24:32.117 26292 26292 W Adjust  : Adjust not initialized, but resetting session partner parameters saved for launch
04-22 21:24:32.118 26292 26292 W Adjust  : SANDBOX: Adjust is running in Sandbox mode. Use this setting for testing. Don't forget to set the environment to `production` before publishing!
04-22 21:24:32.120 26292 26292 W Adjust  : SANDBOX: Adjust is running in Sandbox mode. Use this setting for testing. Don't forget to set the environment to `production` before publishing!
04-22 21:24:32.123 26292 26292 D Adjust  : GoogleAdId being read in the foreground
04-22 21:24:32.127 26292 26343 D Adjust  : Read Attribution: tt:51dilf6 tn:Organic net:Organic cam: adg: cre: cl: adid:e4b48e1722dddcc9c625d8d78cfef2df ct: ca:NaN cc:
04-22 21:24:32.130 26292 26343 D Adjust  : Read Activity state: ec:5 sc:1 ssc:6 sl:1078.9 ts:970.4 la:11:12:13 uuid:ef1bbf79-2289-4a2b-82ad-bbe86c36383d
04-22 21:24:32.131 26292 26343 D Adjust  : Read Session Callback parameters: null
04-22 21:24:32.131 26292 26343 D Adjust  : Read Session Partner parameters: null
04-22 21:24:32.132 26292 26343 D Adjust  : adjust_config.properties file not found in this app
04-22 21:24:32.450 26292 26345 D Adjust  : GoogleAdId read 52f3897a-8f5c-4a7c-9691-9461ca24a8b0
04-22 21:24:32.452 26292 26343 I Adjust  : Google Play Services Advertising ID read correctly at start time
04-22 21:24:32.453 26292 26343 V Adjust  : Foreground timer configured to fire after 60.0 seconds of starting and cycles every 60.0 seconds
04-22 21:24:32.453 26292 26343 I Adjust  : Send in background configured
04-22 21:24:32.458 26292 26343 W Adjust  : Session Callback parameters are not set
04-22 21:24:32.458 26292 26343 D Adjust  : Wrote Session Callback parameters: null
04-22 21:24:32.458 26292 26343 W Adjust  : Session Partner parameters are not set
04-22 21:24:32.459 26292 26343 D Adjust  : Wrote Session Partner parameters: null
04-22 21:24:32.459 26292 26347 D Adjust  : Read Package queue: []
04-22 21:24:32.459 26292 26347 D Adjust  : Package handler read 0 packages
--------- beginning of system

**Comment by Ashvini135** [comment]
Hi [uerceg](https://github.com/uerceg),

Thank you so much for your big, but we are still struggling with fetching the params from adjust link, our "LaunchReceivedDeeplink" never be hitted we test with  some Toast Messages as well inside the LaunchReceivedDeeplink method ,  but in both Environment(Sandbox/Production)  that toast message is not displayed.

**Comment by uerceg** [comment]
Based on these logs from above, I can see that before you opened the app, it was already installed. All the steps from [this comment](https://github.com/adjust/android_sdk/issues/397#issuecomment-1517712344) are important to be followed, but from the logs you sent me it doesn't appear as if steps 4 and 5 were done.

**Comment by doolong189** [comment]
To get the referral code value from deeplink from Adjust, do I need to set anything on Adjust and add code command to get that value? Android java/kotlin. Please help me.
